pkg/trace/writer/trace.go

```diff
@@ -139,6 +144,49 @@ func (w *TraceWriter) Run() {
 	}
 }
 
+func (w *TraceWriter) runSync() {
+	defer close(w.stop)
+	defer close(w.flushChan)
+	for {
+		select {
+		case pkg := <-w.In:
+			w.addSpans(pkg)
+		case <-w.flushChan:
+			// Make sure all pending spans have been buffered
```

Comment:
I think we can leave this comment out. The method name on the next line is very selfexplanatory.